lib/order.rb

```diff
@@ -0,0 +1,43 @@
+#ordering
+require_relative 'takeaway'
+require_relative 'confirm'
+
+class Order 
+
+  attr_reader :basket, :takeaway, :total
+
+  def initialize(takeaway = Takeaway) 
+    @basket = []
+    @takeaway = takeaway.new
+    @text = Text.new
+    @selection = {}
+    @total = 0
+  end 
+
+  def view
+    @takeaway.view
+  end 
+
+  def select_dish(dish, quantity, price)
+    @selection[dish] = quantity
```

Comment:
'selection' could potentially be a hash defined inside the method rather than an instance variable, and passed through to add_to_basket(selection). Then you won't need to nil it after each use